packages/migration/src/migrations/v0011_pageAdvancement.ts

```diff
@@ -45,22 +45,15 @@ export type UnionElementV11 =
 
 export interface StoryV11 extends Omit<StoryV10, 'pages'> {
   pages: PageV11[];
-  autoAdvance: boolean;
-  defaultPageDuration: number;
+  autoAdvance?: boolean;
+  defaultPageDuration?: number;
 }
 export interface PageV11 extends Omit<PageV10, 'elements'> {
   elements: UnionElementV11[];
 }
 
 function pageAdvancement(story: StoryV10): StoryV11 {
-  if ('autoAdvance' in story && 'defaultPageDuration' in story) {
-    return story as StoryV11;
-  }
-  return {
-    autoAdvance: true,
-    defaultPageDuration: 7,
-    ...story,
-  };
+  return story as StoryV11;
```

Comment:
Have to remove always setting the auto-advance and page duration, otherwise we can't really detect if it needs to be overridden by global settings.

Reply:
@merapi FYI I stepped through the code locally and looks like the `version` is `undefined` for a new story, meaning that every new story will go through the full migration script initially 🤷 Not sure if that's the intended way or a bug.

Reply:
@swissspidy Perhaps you have thoughts about this?
Looks like new stories will always go through all migrations. Since V11 adds auto-advance props but we shouldn't have those by default on new stories to use the global ones instead, we don't need V11 migration anymore. Should we just remove it? Is it a bug that the migration runs through new stories?

Reply:
> Looks like new stories will always go through all migrations. Is it a bug that the migration runs through new stories?

AFAIK it's intentional because this way a new story automatically gets the right and up-to-date data structure.

We shouldn't modify existing migrations though, since this won't work for existing stories.

It'd be better to add a new migration, e.g. to delete `defaultPageDuration` if it's `7`.

Reply:
> It'd be better to add a new migration, e.g. to delete defaultPageDuration if it's 7.

Yep, I wasn't sure about what's the best way to proceed with that -- removed the current migration to make it work functionally.

If we add a new migration, the existing stories with the default values would be changed and this would mean the new global values would influence those stories, too, not just the new stories. If that's ok then this sounds like a good plan.

Reply:
Not necessarily. The new global value will be default indeed, however, if the user first changes the global value, and then opens an existing story with previous default values, it will use the new global values for the story -- since the migration runs only if a story is opened, and it can be opened after the global value has been changed.